src/modules/services/submodule/buySiriusCoins/views/ViewServicesStableCoinsCheckStatus.vue

```diff
@@ -0,0 +1,201 @@
+import { useRoute } from "vue-router";
+import { networkState } from "@/state/networkState";
+import { ChainSwapConfig } from "@/models/stores/chainSwapConfig";
+import { useI18n } from 'vue-i18n';
+import { SwapUtils } from '@/util/swapUtils';
+import { Helper } from "@/util/typeHelper";
+
+
+export default {
+  name: "ViewServicesStableCoinsCheckStatus",
+
+  setup(){
+
+    const {t} = useI18n();
+    const route = useRoute();
+
+    const isLoaded = shallowRef(false);
+
+    let swapData = new ChainSwapConfig(networkState.chainNetworkName);
+    swapData.init();
+
+    const customErrorMessage = ref('');
+    const transactionHash = ref('');
+    const hashType = ref('BSC');
+
+    const disabledCheckStatus = computed(() => {
+      if(isLoaded.value){
+        return true;
+      }
+      if(hashType.value == 'BSC'){
+        // check for BSC txn type
+        if(transactionHash.value.length == 66){
+          if(transactionHash.value.substring(0, 2).toLocaleUpperCase() == '0X'){
+            return false
+          }
+          return true;
+        }else{
+          return true;
+        }
+      }else{
+        // check for sirius hash
+        if(transactionHash.value.length == 64){
+          return false;
+        }else{
+          return true;
+        }
+      }
+    });
+
+    const isSuccess = shallowRef(false);
+
+    const amount = ref('');
+    const siriusAddress = ref('');
+    const siriusTransactionHash = ref('');
+    const remoteTxnHash = ref('');
+    const txnStatus = ref('');
+    const time = ref('');
+    const explorerLink = () =>{
+    if(!networkState.currentNetworkProfile){
+        return ''
+    }
+    return networkState.currentNetworkProfile.chainExplorer.url + '/' + networkState.currentNetworkProfile.chainExplorer.hashRoute + '/';
+}
+    const remoteExplorerLink = ref('');
+    remoteExplorerLink.value = hashType.value === 'BSC'?swapData.BSCScanUrl:swapData.ETHScanUrl;
+
+    const checkStatus = async() => {
+      isLoaded.value = true;
+      isSuccess.value = false;
+      customErrorMessage.value = '';
+
+      try {
+        let url = '';
+        if(hashType.value == 'BSC'){
+          url = SwapUtils.getIncoming_BSCBuySiriusTokenCheckRemoteStatus_URL(swapData.swap_IN_SERVICE_URL, 'bsc', transactionHash.value);
+        }else{
+          url = SwapUtils.getIncoming_BSCBuySiriusTokenCheckStatus_URL(swapData.swap_IN_SERVICE_URL, transactionHash.value);
+        }
+        const response = await fetch(url, {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+
+        if(response.status == 200){ // data.status == 'fulfilled'
+        const json = await response.json();
+          remoteTxnHash.value = json.remoteTxnHash;
+          siriusTransactionHash.value = json.siriusTxnHash;
+          siriusAddress.value = Helper.createAddress(json.siriusAddress).pretty();
+          txnStatus.value = json.status.toLocaleUpperCase();
```

Comment:
why use toLocaleUpperCase ? or you mean toUpperCase ? 
txnStatus maybe need to take Sirius chain check status into consideration.
If there is siriusTxnHash, just check the status of it from Sirius, and display the status of it.

Reply:
Yeah, should be toUpperCase, seems both give the same result. Will amend. 

If the status from the api is only for the remote hash, will check with sirius sdk for the status too.

Reply:
The swap server is focus on the swap only, you can use the ready AppState to do it